test/test_sdk_installation.py

```diff
@@ -13,44 +13,78 @@
 import subprocess
 import sys
 
+# Constants
+
+SCRIPTS_PATH = os.path.dirname(__file__) + "/scripts/"
+
+# Helper functions
+
+def scripts_for(path):
+    return [f for f in os.listdir(SCRIPTS_PATH + path) if f.endswith(".py")]
+
+
+def scripts_path_on_vm(path):
+    return "/vagrant/test/scripts/" + path + "/"
+
+
+def run_test_script_inside_vm(script_type, script_name, script_args=""):
+    command_to_run_in_vm = "python3 {}/{} {}".format(
+        scripts_path_on_vm(script_type),
+        script_name,
+        script_args)
+
+    args = ["vagrant", "ssh", "-c", command_to_run_in_vm]
+    process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    script_exit_code = process.wait()
+
+    if script_exit_code != 0:
+        print(pipe_to_str(process.stdout))
+        print(pipe_to_str(process.stderr), file=sys.stderr)
+
+    assert script_exit_code == 0
+
+
+def pipe_to_str(pipe):
+    return "\n".join([ x.decode() for x in pipe.readlines() ])
+
+
+# Fixtures
 
 @pytest.fixture(scope="class")
 def vagrant():
     try:
-        assert os.system("SDK_FILE_NAME=\"stub_sdk.sh\" NO_GUI=1 vagrant up") == 0
+        assert os.system("SDK_FILE_NAME=\"test/stubs/full_sdk.sh\" NO_GUI=1 vagrant up") == 0
         yield
     finally:
         os.system("vagrant destroy -f")
 
 
+@pytest.fixture(scope="class")
+def vagrant_no_qt():
+    try:
+        assert os.system("SDK_FILE_NAME=\"test/stubs/sans_qmake_sdk.sh\" NO_GUI=1 vagrant up") == 0
+        yield
+    finally:
+        os.system("vagrant destroy -f")
```

Comment:
Would be nice if the try finally block were extracted to a function that gets called instead? then the what stub_sdk could be a parameter to that function.

Example
```python
def do_vagrant_up(sdk_file_name):
    try:
        assert os.system('SDK_FILE_NAME="test/stubs/{}" NO_GUI=1 vagrant up'.format(sdk_file_name)) == 0
        yield
    finally:
        os.system("vagrant destroy -f")

@pytest.fixture(scope="class")
def vagrant():
    do_vagrant_up("full_sdk.sh")

@pytest.fixture(scope="class")
def vagrant_no_qt():
    do_vagrant_up("sans_qmake_sdk.sh")
```

Reply:
I tried this but for some reason, or most probably because I don't quite understand the usage of yield and assert in this, it doesn't work like you proposed, as soon as I move the assert und yield into the other function the tests fail and tell me that I should run `vagrant up`. I think I will leave it like it is for now.